### ISSUE TITLE
Scroll to the start node of the first text directive instead of its common ancestor

### ISSUE THREAD

**Issue by jnjaeschke** [WICG/scroll-to-text-fragment#259]
If a text directive should be scrolled to, it should be visible [in the center of the view](https://wicg.github.io/scroll-to-text-fragment/#invoking-text-directives:~:text=range%2C%20then%3A-,Set%20target%20to%20be%20the%20first%20common%20ancestor%20of%20target%E2%80%99s%20start%20node%20and%20end%20node.,-While%20target%20is&text=Let%20blockPosition%20be%20%22center%22%20if%20scrollTarget%20is%20a%20range%2C%20%22start%22%20otherwise.). However, scrolling the common ancestor of a range into the center of the view is not necessarily scrolling the range into view.

Example:

```html
<html>
  <style>
    .space {
      margin-top: 200vh;
    }
  </style>
  <body>
    <p class="space">Test</p>
    <p class="space">Test2</p>
  </body>
</html>
```

Text Directive: `:~:text=Test,Test2`

An implementation that follows the spec would scroll in between the two `<p>`, so that neither `Test` nor `Test2` would be visible. This is confusing for users. Therefore, my proposal is to change the spec so that the _start container_ of the range is visible at the center of the view instead of the common ancestor.

Chrome already follows this proposal and scrolls `Test` into the center of the view. Safari and Gecko (as of now) follow the spec. I'm planning to [change the implementation in Gecko](https://bugzil.la/1906895) to align with Chrome.

/cc @zcorpan

**Comment by bokand** [comment]
Yeah, I agree that makes more sense. The intent here was that the first common ancestor is used to set the `:target` pseudo class since we can only set it on one element (I think?). But scrolling to it seems like a bug - we could separate the element used for `:target` and scroll the start container as you suggest.

**Comment by annevk** [comment]
cc @megangardner

**Comment by mrego** [comment]
I've noticed a difference between Chrome and Firefox behavior. Chrome centers the top of the element, while Firefox and WebKit centers the element. Example:
```
data:text/html,<!DOCTYPE html><div style="margin-block: 200vh; font-size: 300px;">test</div>#:~:text=test
```

This also happens when the element is taller, like the initial example on this issue, or the following one:
```
data:text/html,<!DOCTYPE html><div style="margin-block: 200vh; font-size: 300px;">start<br><br><br><br>end</div>#:~:text=start,end
```

However horizontally the centering is not happening in any browser:
```
data:text/html,<!DOCTYPE html><div style="margin-inline: 200vh; font-size: 300px;">test</div>#:~:text=test
```

Maybe something like [`safe center`](https://drafts.csswg.org/css-align/#overflow-values) alignmnet would be the desired behavior here.
